### PR TITLE
Upgrade Guava version to 18.0

### PR DESCRIPTION
**Pull request by HongW2019**
## What changes were proposed in this pull request?

(Upgrade Guava version to 18.0)

## How was this patch tested?

(Passed unit tests, different scales' TPC-DS)

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @lidinghao ~

**Comment by LuciferYang** [comment]
Will it cause the problem of guava version compatibility? We know the compatibility of guava itself is not very well, another way is to use `caffeine cache` instead of `guava cache`.

I've done a simple test with caffeine [cache-use-caffeine](https://github.com/Intel-bigdata/OAP/compare/master...LuciferYang:cache-use-caffeine?expand=1)

**Comment by HongW2019** [comment]
> @HongW2019
> 
> Will it cause the problem of guava version compatibility? We know the compatibility of guava itself is not very well, another way is to use `caffeine cache` instead of `guava cache`.
> 
> I've done a simple test with caffeine [cache-use-caffeine](https://github.com/Intel-bigdata/OAP/compare/master...LuciferYang:cache-use-caffeine?expand=1)

@LuciferYang 

So far, after having finished different scale's TPC-DS, the guava 18.0 hasn't encountered any compatibility error. Although we used maven shade plugin to the guava dependency in OAP,  maybe we should add more test scenarios, do you have any suggestions to test compatibility of guava?

**Comment by lidinghao** [comment]
Look good to me

**Comment by LuciferYang** [comment]
@HongW2019 I think we should remove guava-xxx.jar from the jars directory when do `TPC-DS Test` to ensure Executor always load guava code from our specific version.

**Comment by LuciferYang** [comment]
@HongW2019 I give a pr #1134 to use caffeine cache instead of guava cache, we can discuss this offline, and cc @lidinghao  @zhixingheyi-tian ~

File: src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala [hunk 1]
```diff
@@ -304,10 +303,10 @@ class GuavaOapCache(
           if (fiber.isInstanceOf[DataFiberId] || fiber.isInstanceOf[TestDataFiberId]) {
             val fiberCache = cacheInstance.get(fiber)
             // Avoid loading a fiber larger than DATA_MAX_WEIGHT / CONCURRENCY_LEVEL
-            assert(fiberCache.size() <= DATA_MAX_WEIGHT * KB / CONCURRENCY_LEVEL,
+            assert(fiberCache.size() <= DATA_MAX_WEIGHT / CONCURRENCY_LEVEL,
```
Comment (LuciferYang) [inline src/main/scala/org/apache/spark/sql/execution/datasources/oap/filecache/OapCache.scala]: Actually I think use OccupiedSize not size in CacheView, Maybe we should  another pr to fix this @HongW2019

**Comment by HongW2019** [comment]
> @HongW2019 I think we should remove guava-xxx.jar from the jars directory when do `TPC-DS Test` to ensure Executor always load guava code from our specific version.
@LuciferYang  As we know Shade plugin can rename guava dependency and then create a shaded jar, which can help ensure Executor always load guava code from our specific version.  About removing guava-xxx.jar from jars, did you mean removing it from $SPARK_HOME/jars? which would encounter errors when launch Spark .

**Comment by LuciferYang** [comment]
@HongW2019 right,  removing it from $SPARK_HOME/jars.
If an error is reported, Maybe it means that we are still use this version instead of the 18.0 specified by us

**Pull request closed without merging**